src/Service/QuizInstanceService.php

```diff
@@ -22,9 +22,15 @@ class QuizInstanceService extends AbstractService
     public function addQuiz(int $userid, int $quizid) {
         $quiz = new QuizInstance();
         $template = $this->repoManager->getRepository(QuizTemplate::class)->find($quizid);
-        if ($template == null) {
+
+        if (!$template) {
             return null;
         }
+
+        if ($this->findByUser($userid, $quizid)) {
+            return $this->findByUser($userid, $quizid)->getID();
```

Comment:
you're doing an unnecessary call, one in if, one inside if. move call outside if in a variable and then do the check and get the id